fs/copy_linux.go

```diff
@@ -49,22 +49,31 @@ func copyFileContent(dst, src *os.File) error {
 		return errors.Wrap(err, "unable to stat source")
 	}
 
-	n, err := unix.CopyFileRange(int(src.Fd()), nil, int(dst.Fd()), nil, int(st.Size()), 0)
-	if err != nil {
-		if err != unix.ENOSYS && err != unix.EXDEV {
-			return errors.Wrap(err, "copy file range failed")
-		}
+	var written int64
+	size := st.Size()
 
-		buf := bufferPool.Get().(*[]byte)
-		_, err = io.CopyBuffer(dst, src, *buf)
-		bufferPool.Put(buf)
-		return err
-	}
+	first := true
+	for written < size {
+		n, err := unix.CopyFileRange(int(src.Fd()), nil, int(dst.Fd()), nil, int(size-written), 0)
```

Comment:
you shouldnt cast `size-written` to int, it could be larger. you will need a check for is it >2G and then write a smaller amount given the length arg is an int.

Reply:
(the syscall doesnt take an int here so in C you would be able to pass a 64 bit value)

Reply:
Fixed.